Python/compile.c

```diff
@@ -1655,16 +1655,10 @@ compiler_body(struct compiler *c, location loc, asdl_stmt_seq *stmts)
 static int
 compiler_codegen(struct compiler *c, mod_ty mod)
 {
-    _Py_DECLARE_STR(anon_module, "<module>");
-    RETURN_IF_ERROR(
-        compiler_enter_scope(c, &_Py_STR(anon_module), COMPILER_SCOPE_MODULE,
-                             mod, 1));
-
     location loc = LOCATION(1, 1, 0, 0);
     switch (mod->kind) {
     case Module_kind:
         if (compiler_body(c, loc, mod->v.Module.body) < 0) {
-            compiler_exit_scope(c);
```

Comment:
I'm a bit confused by this; how does this not leave us with one fewer `compiler_exit_scope` than we had previously? We still have the same number of `compiler_enter_scope` (it's just moved from above into both callers of `compiler_codegen`), and both callers still do the same number of `compiler_exit_scope` that they did before (1), but we eliminate this `compiler_exit_scope`. Why?

Reply:
I think currently we may have one too many - one exit here and then we return to the caller which exits again.

Reply:
Actually I'm wrong in the case of compiler_mod, I'll fix it there.